torchqdynamics/utils/tensor_types.py

```diff
@@ -99,11 +99,25 @@ def wrapper(
     return wrapper
 
 
+DTYPE_CONVERSION_TABLE = {
+    torch.float32: torch.complex64,
+    torch.float64: torch.complex128,
+    torch.complex64: torch.float32,
+    torch.complex128: torch.float64,
+}
+
+
 def dtype_complex_to_float(
     dtype: torch.complex64 | torch.complex128 | None = None,
 ) -> torch.float32 | torch.float64:
     dtype = cdtype(dtype)
-    return torch.float64 if dtype is torch.complex128 else torch.float32
+    return DTYPE_CONVERSION_TABLE[dtype]
+
+
+def dtype_float_to_complex(
+    dtype: torch.float32 | torch.float64 | None = None,
+) -> torch.complex64 | torch.complex128:
+    return DTYPE_CONVERSION_TABLE[dtype]
```

Comment:
Fails if `None` is passed.

Reply:
It's fixed. I also updated the name from `dtype_float_to_complex` to `dtype_real_to_complex` everywhere, to match the naming convention of pytorch.